<launch>

    <!-- Robot name should add a slash -->
    <arg name="robot_name" default=""/>
    <arg name="robot_name_slash" default=""/>

    <node pkg="odometry" type="odometry_node" name="odometry_node" output="screen">
        
        <param name="active" value="true"/>
        <param name="publish" value="true"/>

        <param name="twist_topic" value="/$(arg robot_name)/Toposition"/>
        <param name="odom_topic" value="/$(arg robot_name)/odom"/>
        <remap from="/cmd_vel" to="/$(arg robot_name)/cmd_vel"/>
        <remap from="/ekf_pose" to="/$(arg robot_name)/ekf_pose"/>

		<param name="fixed_frame" value="$(arg robot_name_slash)odom"/>
        <param name="target_frame" value="$(arg robot_name_slash)base_footprint"/>
        <param name="update_params" value="false"/>
        <param name="covariance_x" value="0.05"/>
        <param name="covariance_y" value="0.05"/>
        <param name="covariance_z" value="0.1"/>
        <param name="covariance_vx" value="0.05"/>
        <param name="covariance_vy" value="0.05"/>
        <param name="covariance_vz" value="0.2"/>
        <param name="covariance_multi_vx" value="0.5"/>
        <param name="covariance_multi_vy" value="0.5"/>
        <param name="covariance_multi_vz" value="0.5"/>
        <param name="using_nav_vel_cb" value="false"/>
        <param name="using_dynamic_reconf" value="false"/>
        <param name="use_stm_integral" value="true"/>

    </node>

</launch>
